### ISSUE TITLE
unity-extra/indicator-psensor fails to build

### ISSUE THREAD

**Issue by renegart** [shiznix/unity-gentoo#238]
Since a few days 'unity-extra/indicator-psensor' fails to build because it doesn't find `bits.h`.
Seems one of a dependencies has removed it.

`x86_64-pc-linux-gnu-gcc -DPACKAGE_DATA_DIR=\"/usr/share/psensor\" -DLOCALEDIR=\"/usr/share/locale\" -DHAVE_CONFIG_H -I. -I..  -Wall -DDEFAULT_WWW_DIR=\""/usr/share/psensor/www"\" -DDATADIR=\""/usr/share"\" -DPSENSOR_DESKTOP_FILE=\""psensor.desktop"\" -I../src/lib -I../src/unity -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DGTK_DISABLE_SINGLE_INCLUDES  -DGDK_DISABLE_DEPRECATED  -DGSEAL_ENABLE   -I/usr/include/libgtop-2.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/libappindicator3-0.1 -I/usr/include/libdbusmenu-glib-0.4 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include    -pthread -I/usr/include/unity/unity -I/usr/include/dee-1.0 -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libdbusmenu-glib-0.4  -I/usr/include/json-c   -march=native -O2 -pipe -c -o ui_unity.o ui_unity.c
x86_64-pc-linux-gnu-gcc -DPACKAGE_DATA_DIR=\"/usr/share/psensor\" -DLOCALEDIR=\"/usr/share/locale\" -DHAVE_CONFIG_H -I. -I..  -Wall -DDEFAULT_WWW_DIR=\""/usr/share/psensor/www"\" -DDATADIR=\""/usr/share"\" -DPSENSOR_DESKTOP_FILE=\""psensor.desktop"\" -I../src/lib -I../src/unity -pthread -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -DGTK_DISABLE_SINGLE_INCLUDES  -DGDK_DISABLE_DEPRECATED  -DGSEAL_ENABLE   -I/usr/include/libgtop-2.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -pthread -I/usr/include/libappindicator3-0.1 -I/usr/include/libdbusmenu-glib-0.4 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib64/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/fribidi -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include    -pthread -I/usr/include/unity/unity -I/usr/include/dee-1.0 -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/lib64/libffi/include -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/libdbusmenu-glib-0.4  -I/usr/include/json-c   -march=native -O2 -pipe -c -o rsensor.o rsensor.c
rsensor.c:23:10: fatal error: bits.h: No such file or directory
   23 | #include <bits.h>
      |          ^~~~~~~~
compilation terminated.
make[3]: *** [Makefile:564: rsensor.o] Error 1
make[3]: Leaving directory '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/work/psensor-1.1.5/src'
make[2]: *** [Makefile:623: all-recursive] Error 1
make[2]: Leaving directory '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/work/psensor-1.1.5/src'
make[1]: *** [Makefile:523: all-recursive] Error 1
make[1]: Leaving directory '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/work/psensor-1.1.5'
make: *** [Makefile:422: all] Error 2
 * ERROR: unity-extra/indicator-psensor-1.1.5_p_p0101_p01::unity-gentoo failed (compile phase):
 *   emake failed
 * 
 * If you need support, post the output of `emerge --info '=unity-extra/indicator-psensor-1.1.5_p_p0101_p01::unity-gentoo'`,
 * the complete build log and the output of `emerge -pqv '=unity-extra/indicator-psensor-1.1.5_p_p0101_p01::unity-gentoo'`.
 * The complete build log is located at '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/temp/environment'.
 * Working directory: '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/work/psensor-1.1.5'
 * S: '/var/tmp/portage/unity-extra/indicator-psensor-1.1.5_p_p0101_p01/work/psensor-1.1.5'
`

**Comment by renegart** [comment]
seems 'dev-libs/json-c' has changed it's interface.
new 'dev-libs/json-c-0.14' is needed by 'sys-fs/cryptsetup-2.3.2'

**Comment by c4pp4** [comment]
Hi, I'm not using it but I think you can try to remove:
`epatch -p1 "${FILESDIR}/json-c-0.13-is_error-build-fix.diff"`
from ebuild and let it to use eoan version patch: **0001-Update-rsensor.c-drop-is_error-macro-definition.patch:**
```
From 31d75df7c516c5e767e6b8219c2c08a39159d699 Mon Sep 17 00:00:00 2001
From: Matheus Teixeira <matheus.mtxr@gmail.com>
Date: Sun, 30 Jun 2019 08:39:18 +0000
Subject: [PATCH] Update rsensor.c: drop is_error macro definition. New json-c
 13.1 removed bits.h inclusion, deprecating de facto all the definitions
 inside. is_error is defined as "#define is_error(ptr) (ptr == NULL)"

in this case, double checking the same pointer is just useless, remove it.
Also simplify the double obj check in remote_psensor_update because obj can't change between the first and the second check

After writing the patch I discovered a new PR From: Matheus Teixeira <matheus.mtxr@gmail.com>, with really similar content, so I'm giving
credits for authorship to him.

Signed-off-by: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Signed-off-by: Gianfranco Costamagna <locutusofborg@debian.org>
```

**Comment by shiznix** [comment]
Fixed in commit 66f7ef9a4ecf9aaa6cf274724f231f2bcc919be9 thanks :)
